src/tests/fs_tests.rs

```diff
@@ -425,19 +426,21 @@ pub mod fs_tests {
         //Writing into that file's first 9 bytes.
         assert_eq!(cage.write_syscall(fd, str2cbuf("Test text"), 9), 9);
 
-        //Checking if not passing any of the two `MAP_PRIVATE`
-        //or `MAP_SHARED` flags correctly results in `The value
-        //of flags is invalid (neither `MAP_PRIVATE` nor
-        //`MAP_SHARED` is set)` error.
-        assert_eq!(
-            cage.mmap_syscall(0 as *mut u8, 5, PROT_READ | PROT_WRITE, 0, fd, 0),
-            -(Errno::EINVAL as i32)
-        );
+        let mmap_result = cage.mmap_syscall(0 as *mut u8, 5, PROT_READ | PROT_WRITE, 0, fd, 0);
+
+        // If mmap fails (returns -1), check errno for specific error (EINVAL)
+        if mmap_result == -1 {
```

Comment:
We should handle getting the correct errno in the mmap_syscall function actually. So check the return value if its -1 there, and if it is use syscall_error, there should be some examples of this. Then you can just check the value here.

Reply:
fixed in `mmap_syscall `